app.rb

```diff
@@ -15,9 +16,52 @@
 # This class represents a Teacher with attributes like id, name, and age.
 class App
   def initialize
-    @people = []
-    @books = []
-    @rentals = []
+    @people = load_data_from_file('people.json') || []
+    @books = load_data_from_file('books.json') || []
+    @rentals = load_data_from_file('rentals.json', @books, @people) || []
+  end
+
+  def save_data_to_file(filename, data)
+    File.write(filename, JSON.pretty_generate(data))
+  end
+
+  def load_data_from_file(filename, books = [], people = [])
+    return [] unless File.exist?(filename)
+
+    json_data = File.read(filename)
+    return [] if json_data.strip.empty?
+
+    parsed_data = JSON.parse(json_data)
+
+    case filename
+    when 'people.json'
+      parsed_data.map do |person_hash|
+        if person_hash.key?('specialization')
+          Teacher.new(person_hash['age'], person_hash['specialization'], person_hash['name'])
+        elsif person_hash.key?('classroom')
+          classroom = Classroom.new(person_hash['classroom'])
+          Student.new(person_hash['age'], classroom, person_hash['name'])
+        else
+          Person.new(person_hash['age'], person_hash['name'])
+        end
+      end
+    when 'books.json'
+      parsed_data.map { |book_hash| Book.new(book_hash['title'], book_hash['author']) }
+    when 'rentals.json'
+      parsed_data.map do |rental_hash|
+        book = books.find { |b| b.id == rental_hash['book_id'] }
+        person = people.find { |p| p.id == rental_hash['person_id'] }
+        Rental.new(rental_hash['date'], book, person)
+      end
+    else
+      []
+    end
+  end
+
+  def save_all_data
+    save_data_to_file('people.json', @people.map(&:to_h))
+    save_data_to_file('books.json', @books.map(&:to_h))
+    save_data_to_file('rentals.json', @rentals.map(&:to_h))
```

Comment:
- [ ] I reviewed your code and it looks great! You were able to successfully save data in files with JSON. However, I noticed that there seems to be a problem loading data from the files on start-up. I've attached a short screenshot to highlight the issue.

Could you take a look at the screenshot and see if you can fix the bug? I'm confident that you can do it!
![Screenshot from 2023-09-13 17-01-09](https://github.com/rudicarrilloypr/school-library-opp/assets/112261237/7184c335-ac80-4621-a442-c66f520b4f2e)

- [ OPTIONAL ] Additionally I have also noticed on every save, your data ids keeps changing, i would be nice to have a look at it too.

Reply:
@zeff96 that issue doesn't happen to us, I believe maybe you did not install the bundles or dependencies before reviewing the project

Reply:
- [ ] Am sorry about that, If there are extra dependencies to install kindly mention that in your readme file, please

- [ ] Please attach a screenshot of your terminal